autogen/agentchat/groupchat.py

```diff
@@ -22,7 +22,7 @@ class GroupChat:
         in its `function_map`.
     """
 
-    agents: List[Agent]
+    agents: List[ConversableAgent]
```

Comment:
Good catch! For our current GroupChat design, the agents are all Conversable Agent, because we require the `system_message` member variable. 

@sonichi What do you think? This change to "Conversable Agent" is accurate and reasonable. However, do we want to redesign GroupChat so that it will be compatible with Agent?

Reply:
Actually, I thought about this again after lunch. It is still better to use "Agent" here, and we may need to redesign the whole group chat.

For instance, I am going to add "DALLE" as an Agent (inherit from Agent), but it is obviously not Conversable. Then, the DALLE agent could not join the group chat, which is not ideal. It would be the same for Tool Agent, and some other non-conversable agents. @sonichi

Reply:
I think Agent is better for generality.

Reply:
As a user I think it will be nice to have other agents in the group chat.